src/common/function/src/scalars/timestamp/to_unixtime.rs

```diff
@@ -79,6 +91,43 @@ impl Function for ToUnixtimeFunction {
                         .collect::<Vec<_>>(),
                 )))
             }
+            ConcreteDataType::Int64(_) | ConcreteDataType::Int32(_) => {
+                let array = columns[0].to_arrow_array();
+                Ok(Arc::new(Int64Vector::try_from_arrow_array(&array).unwrap()))
+            }
+            ConcreteDataType::Timestamp(ts) => {
+                let array = columns[0].to_arrow_array();
+                let value = match ts {
+                    TimestampType::Second(_) => {
```

Comment:
a bit of duplication here..was trying to extract a generic method but could not find one so far. Your suggestions will be highly appreciated 🙏